src/mpi/datatype/typerep/src/typerep_yaksa_init.c

```diff
@@ -33,11 +33,11 @@ yaksa_type_t MPII_Typerep_get_yaksa_type(MPI_Datatype type)
     switch (type) {
         case MPI_CHAR:
         case MPI_SIGNED_CHAR:
-            yaksa_type = YAKSA_TYPE__CHAR;
+            yaksa_type = YAKSA_TYPE__INT8_T;
             break;
 
         case MPI_UNSIGNED_CHAR:
-            yaksa_type = YAKSA_TYPE__UNSIGNED_CHAR;
+            yaksa_type = YAKSA_TYPE__UINT8_T;
             break;
 
         case MPI_PACKED:
```

Comment:
why this change?

Reply:
According to the MPI standard, `MPI_SIGNED_CHAR` and `MPI_UNSIGNED_CHAR` are `C Integer` and most of the arithmetic operations can be performed on them. But inside yaksa, `YAKSA_TYPE__CHAR` and `YAKSA_TYPE__UNSIGNED_CHAR` are treated as characters (supported operation is `REPLACE`).

Reply:
That seems ultimately seems like a problem with yaksa. C defines char types as integral types, though it does not specify whether `char` is signed or not. Should we fix yaksa to support more operations on char types?

Reply:
How about I do that in the collective PR? Another part of this work is supporting yaksa reduction for collectives.

Reply:
Sure, I think that's fine. I double checked MPI, and it explicitly says `MPI_SIGNED_CHAR` and `MPI_UNSIGNED_CHAR` are treated as integral types. `MPI_CHAR` is not supposed to be used in reduction operations.

> 6.9.3 Signed Characters and Reductions
> The types MPI_SIGNED_CHAR and MPI_UNSIGNED_CHAR can be used in reduction operations. MPI_CHAR, MPI_WCHAR, and MPI_CHARACTER (which represent printable characters) cannot be used in reduction operations.